test/passes/memory-packing_all-features.wast

```diff
@@ -26,12 +26,10 @@
    (i32.const 0)
    (i32.const 0)
   )
-  (data.drop 0)
  )
  (func $bar
   (drop
    (loop (result i32)
-    (data.drop 0)
```

Comment:
I'm not sure this function tests anything useful without the data.drop. Should we just leave the data.drops in these tests?

Reply:
After memory packing optimization, this module does not contain any segments left, so this becomes a validation error. The same for the one more removed `(data.drop 0)` above. Before, it was OK because we converted these `data.drop`s to `unreachable`s, because they tried to drop already dropped segments. But dropping the same segments repeatedly is not a trap anymore, so we shouldn't do that. (So segment 0 existed in the beginning and dropped because it was an active segment, but after transformation it does not exist anymore).

Reply:
Oh, come to think of it, then it'd be better to just delete this function, now that it does not test anything meaningful... I already merged this :( But I'll post a follow-up patch.

Reply:
Wait, are you saying that the memory packing removes the segment but keeps the data.drop around, leading to a validation error? That sounds like a bug in the memory packing pass. It should convert the data.drop to a nop or remove it instead.

Reply:
Hmm, yeah, that sounds like a new bug introduced... Is it worth solving given that you are gonna completely overhaul the pass soon?

Reply:
Filed a bug report anyway: #2535